client/resource_group/controller/metrics.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	namespace             = "resource_manager_client"
+	requestSubsystem      = "request"
+	ruSubsystem           = "resource_unit"
+	tokenRequestSubsystem = "token_request"
+
+	resourceGroupNameLabel = "name"
+)
+
+var (
+	resourceGroupStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: "resource_group",
+			Name:      "status",
+			Help:      "Status of the resource group.",
+		}, []string{resourceGroupNameLabel})
+
+	successfulRequestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: requestSubsystem,
+			Name:      "success",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 4, 8), // 0.001 ~ 40.96
+			Help:      "",
+		}, []string{resourceGroupNameLabel})
+
+	failedRequestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: requestSubsystem,
+			Name:      "fail",
+			Help:      "",
+		}, []string{resourceGroupNameLabel})
+
+	successfulTokenRequestDuration = prometheus.NewHistogram(
```

Comment:
Maybe add some comments on why "success" is a "Histogram" and "failure" is a "counter".